packages/next-auth/src/lib/index.ts

```diff
@@ -59,6 +59,33 @@ export interface NextAuthConfig extends Omit<AuthConfig, "raw"> {
   }
 }
 
+function mergeCookiesAndSetCookiesHeaders(headers: Headers): string {
+  const setCookies = headers.get("set-cookie")
```

Comment:
This should use `.getSetCookie`, which is the standard way of retrieving potentially multiple set-cookie values.